tsfpga/git_simulation_subset.py

```diff
@@ -108,11 +118,71 @@ def _find_diff_vhd_files(self) -> set[Path]:
 
     def _iterate_vhd_file_diffs(self, diffs: Any) -> set[Path]:
         """
-        Return the currently existing files that have been changed (added/renamed/modified)
-        within any of the diffs commits.
+        Return the currently existing VHDL files that have been changed (added/renamed/modified)
+        within any of the ``diffs`` commits.
+
+        Will also try to find VHDL files that depend on generated register artifacts that
+        have changed.
         """
         files = set()
 
+        def add_register_artifacts_if_match(
+            diff_path: Path, module_register_data_file: Path, module: "BaseModule"
+        ) -> None:
+            """
+            Note that Path.__eq__ does not do normalization of paths.
+            If one argument is a relative path and the other is an absolute path, they will not
+            be considered equal.
+            Hence, it is important that both paths are resolved before comparison.
+            """
+            if diff_path != module_register_data_file:
+                return
+
+            re_match = self._re_register_data_filename.match(module_register_data_file.name)
+            assert (
+                re_match is not None
+            ), "Register data file does not use the expected naming convention"
+
+            register_list_name = re_match.group(1)
+            regs_pkg_path = module.register_synthesis_folder / f"{register_list_name}_regs_pkg.vhd"
+
+            # It is okay to add only the base register package, since all other
+            # register artifacts depend on it.
+            # This file will typically not exist yet in a CI flow, so it doesn't make sense to
+            # assert for its existence.
+            files.add(regs_pkg_path)
+
+        for diff_path in self._iterate_diff_paths(diffs=diffs):
+            print(diff_path)
+
+            if diff_path.name.endswith(VHDL_FILE_ENDINGS):
+                files.add(diff_path)
+
+            elif self._modules is not None:
+                for module in self._modules:
+                    module_register_data_file = module.register_data_file
+
+                    if isinstance(module_register_data_file, list):
```

Comment:
I think this worked out nicely and didn't add too much overhead with this recursion. One could even add a list of lists 😝 


We are currently quite non-standard wrt to this, but in our case it's a limitation since a module might have multiple register interfaces.